### PR TITLE
Added command base components. Added commandResult and CommandResultE…

### PR DESCRIPTION
**Pull request by MarisGarrett19**
…xtensions to provide a consistent result representation and lets the developer check if the command was successful or not and it allows error reporting as well.

### CODE REVIEW

File: Bookings/Bookings.Shared/Core/Commands/CommandResult.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+namespace Bookings.Application;
+
+/// <summary>
+///     A class to represent the result of a command.
+/// </summary>
+public sealed class CommandResult
+{
+    public List<string>? Errors { get; private set; }
+
+    public object? Data { get; set; }
```
Comment (stap123) [inline Bookings/Bookings.Shared/Core/Commands/CommandResult.cs:12]: Are you happy to allow any object to be returned from a command?

File: Bookings/Bookings.Domain/Core/DomainActionResponse.cs
Comment (stap123) [inline Bookings/Bookings.Domain/Core/DomainActionResponse.cs:1]: Are you sure you want to call this DomainActionResponse?

**Review by stap123 — CHANGES_REQUESTED** [review]

File: Bookings/Bookings.Domain/Core/Result.cs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    {
+        data = Success
+            ? Data
+            : default;
+
+        return Success;
+    }
+
+    /// <summary>
+    ///     This lets us just return the data as a result from a method and have it automatically turned into a domain action response.
```
Comment (stap123) [inline Bookings/Bookings.Domain/Core/Result.cs:84]: need to update the comment :)

**Review by stap123 — CHANGES_REQUESTED** [review]

**Review by stap123 — APPROVED** [review]

**Comment by MarisGarrett19** [comment]
closes #4

**Pull request merged**